lib/project.ts

```diff
@@ -358,23 +359,77 @@ export class Project implements Project.IProject {
+						this.printValidValuesOfProperty(prop).wait();
+					} else {
+						// '$ appbuilder prop print <PropName>' called inside project dir
+						if(_.has(this.projectData, normalizedPropertyName)) {
+							this.$logger.out(this.projectData[normalizedPropertyName]);
+						} else {
+							this.$errors.fail("Unrecognized project property '%s'", property);
+						}
+					}
+				} else {
+					if(options.validValue) {
+						// 'appbuilder prop print --validValue' called inside project dir
+						_.each(_.keys(schema), propKey => {
+							var prop = schema[propKey];
+							this.$logger.info("  " + propKey);
+							this.printValidValuesOfProperty(prop).wait();
+						});
+					} else {
+						// 'appbuilder prop print' called inside project dir
+						var propKeys: any = _.keys(this.projectData);
+						var sortedProperties = _.sortBy(propKeys, (propertyName: string) => {
+							return propertyName.toLowerCase();
+						});
+						_.each(sortedProperties, (propertyName: string) => this.$logger.out(propertyName + ": " + this.projectData[propertyName]));
+					}
+				}
+			} else {
+				// We'll get here only when command is called outside of project directory and --validValue is specified
+				if(property) {
+					var targetFrameworkIdentifiers = _.values(this.$projectConstants.TARGET_FRAMEWORK_IDENTIFIERS);
+					_.each(targetFrameworkIdentifiers, (targetFrameworkIdentifier: string) => {
+						var projectSchema: IDictionary<any> = this.$jsonSchemaValidator.tryResolveValidationSchema(targetFrameworkIdentifier);
+						var currentProp = _.find(_.keys(projectSchema), key => key === property);
+						if(currentProp) {
+							this.$logger.out("  Project type %s:", targetFrameworkIdentifier);
+							this.printValidValuesOfProperty(projectSchema[currentProp]).wait();
+						}
+					});
+				} else {
+					this.$logger.out(this.$projectPropertiesService.getPropertiesForAllSupportedProjects().wait());
+				}
 			}
+		}).future<void>()();
+	}
 
-			var normalizedPropertyName = this.$projectPropertiesService.normalizePropertyName(property, this.projectData);
+	private printValidValuesOfProperty(property: any): IFuture<void> {
+		return (() => {
+			if(property.description) {
+				this.$logger.info("%s%s", Project.INDENTATION, property.description);
+			}
 
-			if(_.has(this.projectData, normalizedPropertyName)) {
-				this.$logger.out(this.projectData[normalizedPropertyName]);
-			} else {
-				this.$errors.fail("Unrecognized project property '%s'", property);
+			if(property.pattern) {
+				this.$logger.trace("%sDesired pattern is: %s", Project.INDENTATION, property.pattern);
 			}
 
+			var validValues: IPropertyValue[] = this.$projectPropertiesService.getValidValuesForProperty(property).wait();
+			if(validValues && validValues.length > 0) {
+				this.$logger.out("%sValid values:", Project.INDENTATION);
+			}
+			_.forEach(validValues, (val: any) => {
+				this.$logger.out("%s  %s    %s", Project.INDENTATION, val.id, val.description);
```

Comment:
It seems that somethimes `val.description` is `undefined` and we print `undefined` on the console. I got the following output 
`Framework
     Valid values:
       Cordova    undefined` when I execute `appbuilder prop print --validValue`. I suggest to print the `description`  only when we have such.